RecoEgamma/EgammaElectronProducers/plugins/GsfElectronBaseProducer.cc

```diff
@@ -22,131 +21,192 @@
+  desc.add<double>("maxFbremBarrel", 999999999.0);
+  desc.add<double>("maxFbremEndcaps", 999999999.0);
+  desc.add<bool>("isBarrel", false);
+  desc.add<bool>("isEndcaps", false);
+  desc.add<bool>("isFiducial", false);
+  desc.add<bool>("seedFromTEC", true);
+  desc.add<double>("maxTIP", 999999999.0);
+  desc.add<double>("minMVA", -0.4);
+  desc.add<double>("minMvaByPassForIsolated", -0.4);
+
+  // preselection parameters (tracker driven only electrons)
+  desc.add<double>("minSCEtBarrelPflow", 0.0);
+  desc.add<double>("minSCEtEndcapsPflow", 0.0);
+  desc.add<double>("minEOverPBarrelPflow", 0.0);
+  desc.add<double>("minEOverPEndcapsPflow", 0.0);
+  desc.add<double>("maxEOverPBarrelPflow", 999999999.0);
+  desc.add<double>("maxEOverPEndcapsPflow", 999999999.0);
+  desc.add<double>("maxDeltaEtaBarrelPflow", 999999999.0);
+  desc.add<double>("maxDeltaEtaEndcapsPflow", 999999999.0);
+  desc.add<double>("maxDeltaPhiBarrelPflow", 999999999.0);
+  desc.add<double>("maxDeltaPhiEndcapsPflow", 999999999.0);
+  desc.add<double>("hOverEConeSizePflow", 0.15);
+  desc.add<double>("hOverEPtMinPflow", 0.0);
+  desc.add<double>("maxHOverEBarrelPflow", 999999999.0);
+  desc.add<double>("maxHOverEEndcapsPflow", 999999999.0);
+  desc.add<double>("maxHBarrelPflow", 0.0);
+  desc.add<double>("maxHEndcapsPflow", 0.0);
+  desc.add<double>("maxSigmaIetaIetaBarrelPflow", 999999999.0);
+  desc.add<double>("maxSigmaIetaIetaEndcapsPflow", 999999999.0);
+  desc.add<double>("maxFbremBarrelPflow", 999999999.0);
+  desc.add<double>("maxFbremEndcapsPflow", 999999999.0);
+  desc.add<bool>("isBarrelPflow", false);
+  desc.add<bool>("isEndcapsPflow", false);
+  desc.add<bool>("isFiducialPflow", false);
+  desc.add<double>("maxTIPPflow", 999999999.0);
+  desc.add<double>("minMVAPflow", -0.4);
+  desc.add<double>("minMvaByPassForIsolatedPflow", -0.4);
 
   // Corrections
-  desc.add<std::string>("superClusterErrorFunction","EcalClusterEnergyUncertaintyObjectSpecific") ;
-  desc.add<std::string>("crackCorrectionFunction","EcalClusterCrackCorrection") ;
- }
+  desc.add<std::string>("superClusterErrorFunction", "EcalClusterEnergyUncertaintyObjectSpecific");
+  desc.add<std::string>("crackCorrectionFunction", "EcalClusterCrackCorrection");
+
+  desc.add<bool>("ecalWeightsFromDB", true);
+  desc.add<std::vector<std::string>>("ecalRefinedRegressionWeightFiles", {})->setComment(
+          "if not from DB. Otherwise, keep empty");
+  desc.add<bool>("combinationWeightsFromDB", true);
+  desc.add<std::vector<std::string>>("combinationRegressionWeightFile", {})->setComment(
+          "if not from DB. Otherwise, keep empty");
+
+  // regression. The labels are needed in all cases.
+  desc.add<std::vector<std::string>>("ecalRefinedRegressionWeightLabels", {});
+  desc.add<std::vector<std::string>>("combinationRegressionWeightLabels", {});
+
+  // Iso values
+  desc.add<bool>("useIsolationValues", false);
+
+  desc.add<std::vector<std::string>>("SoftElecMVAFilesString", {
```

Comment:
I did that now. Since you requested changes anyway, I took the opportunity and reorganized the cuts configuration a bit, which now takes advantage of the fact that it's repeated twice: for ecal-driven a for tracker-driven.